File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minim=prices[0];
        int maxpro=0;
        int n=prices.size();
        for(int i=0;i<n;i++){
            int cost=prices[i]-minim;
            maxpro=max(maxpro,cost);
            minim=min(minim,prices[i]);
        }
        return maxpro;
        
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_current = max_global = 0

        for i in range(1, len(prices)):
            diff = prices[i] - prices[i - 1]  # Daily profit/loss
            max_current = max(0, max_current + diff)  # Reset if negative
            max_global = max(max_global, max_current)  # Update max profit

        return max_global
